docs/design/locale_fallback_and_negotiation.md

```diff
@@ -0,0 +1,427 @@
+* the list of formatting locales available to ICU4X (I18N)
+
+But neither of these give me perfect behavior.
+
+**1. Pass the ICU4X comprehensive list of locales.**
+
+The problem here is that the matcher might return a locale that is supported by ICU4X, but lacks translation for UI.
+
+So the “panel” locale is declared to be `ro-RO` (because ICU4X has data), dates and times will be Romanian, but because there are no Romanian strings most of the UI will be either Spanish or “the ultimate fallback” (English, or maybe the language of the developer team, or the locale declared “default” for the matcher). It does not matter, the result would be a mixture of Romanian dates + something else.
+
+Latin America Spanish for everything would have been consistent.
+
+**2. Pass the list of translated languages for my application**
+
+In this case the best match will be `es-419`, which is Latin American Spanish.
+
+The locale of the panel would be `es-419`, the UI strings would be `es-419`, and the dates / times / etc. would be formatted as LatAm Spanish.
+
+It is consistent, but we could do better. Spanish Chile dates / times might be different from the more “country-neutral” LatAm Spanish, but we ignored all the country specific differences.
+
+If the locale of the panel would be `es-CL` then we would be able to format dates / times the way the user prefers, and would still show UI strings in `es-419`, because the vertical fallback would take care to go from `es-CL` → `es-419` → `es` (somewhere in the string loader / resource manager / whatever component loads UI strings)
+
+Similar problems for all the languages used in many countries (date formats for “generic” Arabic, French, English, when there are tens of countries using these languages)
+
+I learned that the ICU matcher can return two results, one for the UI locale, one for i18n (formatting). ([`LocaleMatcher.Result.makeResolvedLocale`](https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/util/LocaleMatcher.Result.html#makeResolvedLocale--))
+
+That is, if you pass in:
+
+**Available UI:** [`es-419`, `es`]  
+**Desired:** [`es-CL`]
+
+You can get:
+
+**UI:** `es-419`  
+**i18n:** `es-CL`
+
+This is really nice, but still does not cover some valid situations well enough.
+
+For example Android would return `es-CL` in this case, even for UI. And the resource manager will do the fallback to `es-419` when loading the resources.
+
+The benefit: an application might have very-very few resources for `es-CL`. But they are still present.
+
+Some apps use this to keep things like country specific stuff (GPS coordinates, license agreement). But the bulk of the strings would be `es-419`.
+
+If you tell the resource manager to use `es-419` (the UI locale) then those country specific resources are not accessible.
+
+One option would be to pass every locale that has even one resource as “Available UI”:
+
+**Available UI:** [`es-419`, `es`, `es-CL`, `es-AR`, ...]
```

Comment:
Gripe about Github (no action needed): 

Why does the rendering of these 2 lines differ from the rendering of the document portion ~20 lines above that has the same type of formatting?

Reply:
Silly markdown syntax:
 two newlines = `<p>`
 two or more spaces + one newline = `<br>`

This makes (the number) of trailing spaces significant.
Even worse than Python ;-)

Some markdown engines allow for space + back-slash + newline to mean `<br>`
At least that would make it visible in review / diff tools.
But I have no clue what the GitHub markdown engine does.

Reply:
Anyway... Done